### PR TITLE
[Speed] Adjustments to calculations and modifiers

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Uses info gathered by ~~COCONUTS~~ Cocosolos to adjust speed calculations.
- Makes gear penalties and mazurka/quickening additive to base speed, pre-multipliers
- Adjusts additive modifiers to use a flat number to easily add to base speed.
- Discovered a ridiculously small cap for mazurka and quickening combined, meaning they are indeed additive among them.

In short, gear that boosts your speed was unafected, flee is its own multiplier now and gear with penalties + quickening/mazurka are added before the multipliers are applied

## Steps to test these changes

Just use gear and speed-altering stuff while walking

### CODE REVIEW

File: scripts/enum/mod.lua [hunk 1]
```diff
@@ -233,13 +233,17 @@ xi.mod =
...
-    MOVE_SPEED_OVERIDE              = 169, -- Modifier used to overide regular speed caps. (GM speed and Feast of Swords)
-    MOVE_SPEED_STACKABLE            =  75, -- Gear movement speed penalties, flee bonus, etc.
-    MOVE_SPEED_GEAR_BONUS           =  76, -- Gear movement speed bonuses. DOES NOT STACK with each other, only highest applies.
-    MOVE_SPEED_WEIGHT_PENALTY       =  77, -- For Gravity and curse.
-    MOVE_SPEED_QUICKENING           =  78, -- Jig, spreinter shoes, etc. Only highest of Mazurka OR quickening will take effect.
-    MOVE_SPEED_MAZURKA              =  79, -- Song movement speed. Only highest of Mazurka OR quickening will take effect.
-    MOUNT_MOVE                      = 972, -- % Mount Movement Speed
+
+    -- Movement speed modifiers
+    MOVE_SPEED_OVERIDE              =  169, -- Modifier used to overide regular speed caps. (GM speed and Feast of Swords)
```
Comment (EpicTaru) [inline scripts/enum/mod.lua]: Spelling correction: Overide to Override

File: src/map/modifier.h [hunk 2]
```diff
@@ -273,14 +273,15 @@ enum class Mod
...
     // New movement speed modifiers.
-    MOVE_SPEED_OVERIDE        = 169, // Modifier used to overide regular speed caps. (GM speed and Feast of Swords)
-    MOVE_SPEED_STACKABLE      = 75,  // Gear movement speed penalties, flee bonus, etc.
-    MOVE_SPEED_GEAR_BONUS     = 76,  // Gear movement speed bonuses. DOES NOT STACK with each other, only highest applies.
-    MOVE_SPEED_WEIGHT_PENALTY = 77,  // For Gravity and curse.
-    MOVE_SPEED_QUICKENING     = 78,  // Jig, spreinter shoes, etc. Only highest of Mazurka OR quickening will take effect.
-    MOVE_SPEED_MAZURKA        = 79,  // Song movement speed. Only highest of Mazurka OR quickening will take effect.
-
-    MOUNT_MOVE              = 972, // % Mount Movement Speed
+    MOVE_SPEED_OVERIDE        = 169,  // Modifier used to overide regular speed caps. (GM speed and Feast of Swords)
```
Comment (EpicTaru) [inline src/map/modifier.h]: Spelling correction: Overide to Override

**Review by EpicTaru — CHANGES_REQUESTED** [review]

File: src/map/entities/battleentity.cpp [hunk 3]
```diff
@@ -255,40 +255,44 @@ uint8 CBattleEntity::GetSpeed()
...
+    float weightFactor = std::clamp<float>(1.0f - static_cast<float>(getMod(Mod::MOVE_SPEED_WEIGHT_PENALTY)) / 100.0f, 0.1f, 1.0f);
 
     // We have all the modifiers needed. Calculate final speed.
-    float modifiedSpeed = static_cast<float>(baseSpeed) * std::clamp<float>(1.0f + additiveMods + effectBonus, 0.1f, 1.6f) * (1.0f + gearBonus) * std::clamp<float>(1.0f - weightPenalties, 0.1f, 1.0f);
-
-    outputSpeed = static_cast<uint8>(modifiedSpeed);
-
-    // Set cap.
+    // This MUST BE DONE IN THIS ORDER. Using uint8 data type, we use that to floor.
+    outputSpeed = baseSpeed + additiveMods + effectAdditiveBonus;
```
Comment (cocosolos) [inline src/map/entities/battleentity.cpp]: Think effectAdditiveBonus is applied after multipliers. Plumb boots + raptor mazurka + flee = 65, would be 70 if mazurka was affected by flee.

Comment (Xaver-DaRed) [reply src/map/entities/battleentity.cpp]: I'm ready for another round of maths XD

Comment (cocosolos) [reply src/map/entities/battleentity.cpp]: Just to add, effectAdditiveBonus is after flee bonus and after gear bonus. 20 speed + 24% gear = 24 speed, add chocobo mazurka brings it to 34 not 37. So base + additive, times flee, times gear, then quickening/mazurka.

**Review by WinterSolstice8 — APPROVED** [review]

**Review by cocosolos — APPROVED** [review]
Looks good!

**Review by EpicTaru — APPROVED** [review]

**Comment by cocosolos** [comment]
Small mistake on my part, Bolter's roll actually goes between cheer and gear. Otherwise everything else looks right.
